src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double sideLength;
+
+    public Square(Color color, double sideLength) {
+        super(color);
+        this.sideLength = sideLength;
+    }
+
+    @Override
+    public double getArea() {
+        return sideLength * sideLength;
+    }
+
+    @Override
+    public void print() {
+        System.out.println("Figure: square, area: " + getArea() + " sq. units, side length: "
+                + sideLength + " units, color: " + new ColorSupplier().getRandomColor());

```

Comment:
Directly using `System.out.println` in the class method is not a good practice for larger applications as it couples your business logic with the console output. Consider using a logger or returning the string instead.